src/Documentation/Build_Docs.php

```diff
@@ -0,0 +1,116 @@
+<?php
+namespace Tribe\CLI\Documentation;
+
+use WP_CLI;
+
+class Build_Docs extends Abstract_Doc_Command {
+	/**
+	 * Builds WP PHPDoc json file
+	 *
+	 * @param $args
+	 * @param $assoc_args
+	 * @throws WP_CLI\ExitException
+	 */
+	public function build( array $args = null, array $assoc_args = null ) {
+		if ( ! function_exists( '\WP_Parser\parse_files' ) ) {
+			WP_CLI::error( __( 'Please install and activate WP Parser from https://github.com/WordPress/phpdoc-parser before building documentation.', 'bigcommerce' ) );
+			die;
+		}
+
+		$plugin      = $this->parse_plugin( $args );
+		$output_file = $this->parse_file( $plugin, $assoc_args );
+		$data        = $this->get_data();
+		$json        = json_encode( $data, JSON_PRETTY_PRINT );
+		$result      = file_put_contents( $output_file, $json );
+		WP_CLI::line();
+
+		if ( false === $result ) {
+			WP_CLI::error( sprintf( 'Problem writing %1$s bytes of data to %2$s', strlen( $json ), $output_file ) );
+			die;
+		}
+
+		WP_CLI::success( sprintf( 'Data exported to %1$s', $output_file ) );
+		WP_CLI::line();
+	}
+
+	private function get_data() {
+
+		WP_CLI::line( sprintf( 'Extracting PHPDoc from %1$s. This may take a few minutes...', $this->plugin_dir ) );
+		$files = $this->collect_files();
+
+		if ( $files instanceof \WP_Error ) {
+			WP_CLI::error( sprintf( 'Problem with %1$s: %2$s', $this->plugin_dir, $files->get_error_message() ) );
+			die;
+		}
+
+		$output = \WP_Parser\parse_files( $files, $this->plugin_dir );
+
+		return $output;
+	}
+
+	private function collect_files() {
+		$directory = new \RecursiveDirectoryIterator( $this->plugin_dir, \FilesystemIterator::FOLLOW_SYMLINKS );
+		$filter    = new \RecursiveCallbackFilterIterator( $directory, function ( $current, $key, $iterator ) {
+			// Skip hidden files and directories
+			if ( $current->getFilename()[0] === '.' ) {
```

Comment:
Could possibly make this self-documenting and save ourselves a comment by using `$current->isDot()`?

Reply:
This is skipping any file that starts with `.` in addition to the directories: `.` and `..`

Reply:
Ahh gotcha - makes sense :+1: